modules/tracktion_engine/plugins/external/tracktion_ExternalPlugin.cpp

```diff
@@ -1707,22 +1726,38 @@ bool ExternalPlugin::setBusLayout (juce::AudioChannelSet set, bool isInput, int
 }
 
 //==============================================================================
-juce::String ExternalPlugin::createPluginInstance (const juce::PluginDescription& description)
+void ExternalPlugin::createPluginInstance (const juce::PluginDescription& description)
 {
     jassert (! pluginInstance); // This should have already been deleted!
 
     auto& dm = engine.getDeviceManager();
-
-    juce::String error;
-    pluginInstance = engine.getPluginManager().createPluginInstance (description, dm.getSampleRate(), dm.getBlockSize(), error);
-
-    if (pluginInstance != nullptr)
+    
+    if (requiresAsyncInstantiation (engine, description))
+    {
+        // While instantiation is pending, make sure the plugin is disabled
+        setEnabled (false);
+        engine.getPluginManager().pluginFormatManager
+          .createPluginInstanceAsync (description, dm.getSampleRate(), dm.getBlockSize(),
+                                      [&] (auto _instance, const auto& _err) -> void {
```

Comment:
I don't think this can take a reference as the capture type. If the plugin gets deleted before the callback happens this will wreak havoc. Best to take a weak reference instead.
You can do this with `Selectable::getWeakRef()` and dynamic casting it but I think I'm going to add a wrapper class to do this for you automatically (like juce::Component::SafePointer) very soon so you might want to wait for that.